core/src/main/java/org/apache/accumulo/core/fate/Fate.java

```diff
@@ -88,6 +89,34 @@ public enum TxInfo {
     TX_NAME, AUTO_CLEAN, EXCEPTION, TX_AGEOFF, RETURN_VALUE
   }
 
+  public enum FateOperation {
+    COMMIT_COMPACTION,
+    NAMESPACE_CREATE,
+    NAMESPACE_DELETE,
+    NAMESPACE_RENAME,
+    SHUTDOWN_TSERVER,
+    SYSTEM_SPLIT,
+    TABLE_BULK_IMPORT2,
+    TABLE_CANCEL_COMPACT,
+    TABLE_CLONE,
+    TABLE_COMPACT,
+    TABLE_CREATE,
+    TABLE_DELETE,
+    TABLE_DELETE_RANGE,
+    TABLE_EXPORT,
+    TABLE_IMPORT,
+    TABLE_MERGE,
+    TABLE_OFFLINE,
+    TABLE_ONLINE,
+    TABLE_RENAME,
+    TABLE_SPLIT,
+    TABLE_TABLET_AVAILABILITY;
+
+    public static FateOperation fromThrift(TFateOperation tFateOp) {
```

Comment:
Not sure if you have a test for this or not, but it would be good to add one that verifies that every TFateOperation has a FateOperation value and vice-versa. This would help with identifying additions / deletions in future releases.

Reply:
They are not exactly the same. FateOperation has a few more types than TFateOperation. TFateOperation only includes the types used in thrift. Some operations are performed outside of thrift (for example see CompactionCoordinator.java).

Could probably still write some sort of test for this PR though.

Reply:
It might look a little redundant, but I wonder if FateOperation should be FateOperation(TFateOperation) with `null` being an acceptable value. You could still write a test to ensure that all TFateOperations have an associated FateOperation.

Reply:
I agree a test would be good here, you could write something that iterates over TFateOperation values and verifies there's a matching value in FateOperation as @dlmarion mentioned. Because FateOperation is a super set this would catch any future additions to TFateOperation that were not properly added to FateOperation without having to change the test. There's probably not a good way to test the other way though.

Reply:
Thanks for the suggestions @dlmarion and @cshannon. I will add a test for this on Monday.

Reply:
> I wonder if FateOperation should be FateOperation(TFateOperation) with null being an acceptable value

I think this is a good way to connect the two more explicitly. I added this and the test in 30e4753b8203397821cc22c0f6c18941a639213d